File: Java-BTL/src/Class/Class.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package Class;

/**
 *
 * @author 84397
 */
public class Class {
    private Teacher teacher;
    private Room room;
    private StudentGroup studentGroup;
    private ClassTime classTime;
    private Course course;

    public Class(Teacher teacher, Room room, StudentGroup studentGroup, Course course) {
        this.teacher = teacher;
        this.room = room;
        this.studentGroup = studentGroup;
        this.course = course;
        this.classTime=new ClassTime(course.getNumberOfLesson());
    }

    public Teacher getTeacher() {
        return teacher;
    }

    public void setTeacher(Teacher teacher) {
        this.teacher = teacher;
    }

    public Room getRoom() {
        return room;
    }

    public void setRoom(Room room) {
        this.room = room;
    }

    public StudentGroup getStudentGroup() {
        return studentGroup;
    }

    public void setStudentGroup(StudentGroup studentGroup) {
        this.studentGroup = studentGroup;
    }

    public ClassTime getClassTime() {
        return classTime;
    }

    public void setClassTime(ClassTime classTime) {
        this.classTime = classTime;
    }

    public Course getCourse() {
        return course;
    }

    public void setCourse(Course course) {
        this.course = course;
    }
    
    
}

File: v4/DTO/Class.cs
using ConsoleApp;

using System;

public class Class
{
	private Teacher teacher;
	private Course coruse;
	private StudentGroup studentGroup;
	private Room room;
	private ClassTime classTime;
	public Class(Teacher teacher,Course course,StudentGroup studentGroup,Room room )
	{
		this.teacher = teacher;
		this.studentGroup = studentGroup;
		this.coruse = course;
		this.room = room;
		classTime = new ClassTime(coruse.getNumberOfLesson());
	}

	public Teacher getTeacher() { return this.teacher; }
	public Course getCourse() { return this.coruse; }
	public StudentGroup getStudentGroup() {  return this.studentGroup; }
	public Room GetRoom() { return this.room; }
	public ClassTime GetClassTime() { return this.classTime; }
}
